### PR TITLE
mate-panel's x error handler can conflict with gtk's one

### PR DESCRIPTION
**Pull request by dnk**

### CODE REVIEW

File: libmate-panel-applet/mate-panel-applet.c [hunk 1]
```diff
@@ -2423,55 +2423,6 @@ static void mate_panel_applet_factory_main_finalized(gpointer data, GObject* obj
...
- * component's windows) in effectively random order. Furthermore, if
- * you happened to be drawing on any of those windows when the
- * GtkSocket was destroyed (a common state of affairs), an X error
- * will kill your application.
- *
- * To solve this latter problem, MATE Panel sets up its own X error
- * handler which ignores certain X errors that might have been
- * caused by such a scenario. Other X errors get passed to gdk_x_error
- * normally.
- */
```
Comment (monsta) [inline libmate-panel-applet/mate-panel-applet.c]: I wonder if it's okay to remove all this code, given that comment... Won't X server throw some errors at us like it's described there? On the other hand, this code is 5 years old, and I don't know if it's still relevant for modern X versions.

**Comment by monsta** [comment]
Unfortunately, applets still crash if I apply this commit, at the same place and with the same BadPixmap error...

```
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: (wnck-applet:13747): Gdk-ERROR **: The program 'wnck-applet' received an X Window System 
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: This probably reflects a bug in the program.
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: The error was 'BadPixmap (invalid Pixmap parameter)'.
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: (Details: serial 4850 error_code 4 request_code 54 (core protocol) minor_code 0)
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: (Note to programmers: normally, X errors are reported asynchronously;
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: that is, you will receive the error a while after causing it.
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: To debug your program, run it with the GDK_SYNCHRONIZE environment
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: variable to change this behavior. You can then get a meaningful
дек 28 17:32:44 jessica org.mate.panel.applet.WnckletFactory[13691]: backtrace from your debugger if you break on the gdk_x_error() function.)
дек 28 17:32:44 jessica kernel: traps: wnck-applet[13747] trap int3 ip:7f007ddee88b sp:7ffc236e90d0 error:0
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: (stickynotes-applet:13751): Gdk-ERROR **: The program 'stickynotes-applet' rece
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: This probably reflects a bug in the program.
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: The error was 'BadPixmap (invalid Pixmap parameter)'.
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: (Details: serial 907 error_code 4 request_code 54 (core protocol) minor_code 0)
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: (Note to programmers: normally, X errors are reported asynchronously;
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: that is, you will receive the error a while after causing it.
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: To debug your program, run it with the GDK_SYNCHRONIZE environment
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: variable to change this behavior. You can then get a meaningful
дек 28 17:32:45 jessica org.mate.panel.applet.StickyNotesAppletFactory[13691]: backtrace from your debugger if you break on the gdk_x_error() function.)
дек 28 17:32:45 jessica kernel: traps: stickynotes-app[13751] trap int3 ip:7f349495a88b sp:7ffd85127350 error:
```

**Comment by monsta** [comment]
I've also tried to put ```BadPixmap``` in the list of ignored errors instead:

```diff
diff --git a/libmate-panel-applet/mate-panel-applet.c b/libmate-panel-applet/mate-panel-applet.c
index bd1806a..e4abb53 100644
--- a/libmate-panel-applet/mate-panel-applet.c
+++ b/libmate-panel-applet/mate-panel-applet.c
@@ -2436,7 +2436,8 @@ _x_error_handler (Display *display, XErrorEvent *error)
         * X-server-induced errors. Keeping a list of windows for which we
         * will ignore BadDrawables would be a good idea. */
        if (error->error_code == BadDrawable ||
-           error->error_code == BadWindow)
+           error->error_code == BadWindow ||
+           error->error_code == BadPixmap)
                return 0;
 
        return _x_error_func (display, error);
```

Does nothing...

**Comment by raveit65** [comment]
I attached stacktraces at report and they shows something about BadPixmap

**Comment by raveit65** [comment]
I'm pretty sure that comes from libwnk3 itself, see https://bugzilla.gnome.org/show_bug.cgi?id=767803

**Pull request closed without merging**